pocs/focuser/focuslynx.py

```diff
@@ -294,19 +295,36 @@ def _send_command(self, command_str, expected_reply):
         Utility function that handles the common aspects of sending commands and
         parsing responses.
         """
+        name = self.port
+        if hasattr(self, '_focuser_config'):
```

Comment:
That seems like an odd test for whether or not self.uid is present & valid. Perhaps a name attr is needed, initialized where _focuser_config is first set in (presumably in __init__).

Reply:
@AnthonyHorton can you maybe speak to why it was set up this way in #547? I cleaned it up a bit to assume the port is available and then do the check for the `_focuser_config`, so it is slightly different but functionally the same.  Would we be okay to change the `.name` property to better handle this situation rather than do it here?

Reply:
Yeah, this was a middle of the night fix!  All the changes here stem from receiving extremely unhelpful errors for a situation where the FocusLynx focus controller was powered off. The initialisation would get as far as opening the serial port (because that's just a USB-RS232 converter that *does* have power) but then gets no response to the first command it sends, a request for hub info. At that point I was getting a confusing `AttributeError` about `_focuser_config` being missing because it was trying to put the UID into the error message, and the getter for UID tried to get it from `_focuser_config`, which isn't set until getting the focuser config from the controller with the next command.

I make some quick changes so that the error handling would still produce an error message in the case of an error occurring before `_focuser_config` has been set, which enabled me to discover that the actual problem was that no response was being received at all, so I added another handler for that specific case, then went and checked the power switch on the side of the focus controller...

Bit wary of doing anything with the `.name` property. All the `Camera` and `Focuser` classes have this property and at the moment use it in a fairly consistent manner. Would prefer to avoid any `FocusLynx` specific behaviour with that. Maybe give the UID getter a fallback option where it returns `port` if `_focuser_config` is not set?

Side note: for the FocusLynx focusers the 'UID' is __not__ guaranteed to always be unique. There are no serial numbers accessible to external code, the only available identifier is the user set 'nickname' which is saved in the focus controller, and that can be any string the user wants. I don't think non-unique IDs would actually cause problems for focusers (unlike cameras, where it would mess things up) but something to watch out for.